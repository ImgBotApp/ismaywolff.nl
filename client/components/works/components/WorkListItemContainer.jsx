import React, { PropTypes } from 'react'
import { connect } from 'react-redux'
import WorkListItem from './WorkListItem'
import { getEntities } from '../selectors'

export function WorkListItemContainer({ id, entities }) {
  return <WorkListItem work={entities.works[id]} />
}

WorkListItemContainer.propTypes = {
  id: PropTypes.string.isRequired,
  entities: PropTypes.shape({
    works: PropTypes.object.isRequired
  }).isRequired
}

const mapStateToProps = state => ({
  entities: getEntities(state)
})

export default connect(mapStateToProps)(WorkListItemContainer)
