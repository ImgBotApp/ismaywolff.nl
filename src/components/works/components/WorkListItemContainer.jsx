import React, { PropTypes } from 'react'
import { connect } from 'react-redux'
import { NAME } from '../constants'
import WorkListItem from './WorkListItem'

export function WorkListItemContainer({ id, works }) {
  return <WorkListItem work={works.entities[id]} />
}

WorkListItemContainer.propTypes = {
  id: PropTypes.string.isRequired,
  works: PropTypes.shape({
    entities: PropTypes.object.isRequired
  }).isRequired
}

const mapStateToProps = state => ({
  works: state[NAME]
})

export default connect(mapStateToProps)(WorkListItemContainer)
