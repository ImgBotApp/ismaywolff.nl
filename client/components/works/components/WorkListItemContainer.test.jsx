import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { WorkListItemContainer } from './WorkListItemContainer'

describe('<WorkListItemContainer />', () => {
  it('renders correctly', () => {
    const works = {
      1: {
        description: 'description',
        published: 'published',
        slug: 'title',
        title: 'title',
        type: 'type'
      }
    }
    const wrapper = shallow(<WorkListItemContainer id={'1'} works={works} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
