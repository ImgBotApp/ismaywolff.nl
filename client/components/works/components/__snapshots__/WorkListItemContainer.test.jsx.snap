// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkListItemContainer /> renders correctly 1`] = `
<WorkListItem
  work={
    Object {
      "description": "description",
      "published": "published",
      "slug": "title",
      "title": "title",
      "type": "type",
    }
  }
/>
`;
